pylabrobot/liquid_handling/backends/hamilton/STAR.py

```diff
@@ -2476,8 +2478,17 @@ async def iswap_release_picked_up_resource(
 
     assert self.iswap_installed, "iswap must be installed"
 
-    # Get center of source plate. Also gripping height and plate width.
-    center = location + resource.rotated(z=rotation).center() + offset
+    # Get center of source plate in absolute space.
+    # The computation of the center has to be rotated so that the offset is in absolute space.
+    center_in_absolute_space = Coordinate(*matrix_vector_multiply_3x3(
+      resource.rotated(z=rotation).get_absolute_rotation().get_rotation_matrix(),
+      resource.center().vector()
+    ))
+    # This is when the resource is rotated (around its origin), but we also need to translate
+    # so that the lfb of the plate is lfb in absolute space, not local.
```

Comment:
what is lfb? worth spelling it out. abbreviations are generally a code smell

Reply:
left front bottom, used quite often in PLR but will spell out if better

Reply:
actually that's a lie, we dont write that in the repo. but i use it on gh often. will spell out